pkg/splunk/enterprise/configuration.go

```diff
@@ -606,3 +609,63 @@ func updateSplunkPodTemplateWithConfig(podTemplateSpec *corev1.PodTemplateSpec,
 		podTemplateSpec.Spec.Containers[idx].Env = env
 	}
 }
+
+// ListSmartStoreVolumes lists smartstore volumes
+func ListSmartStoreVolumes(volumeList []enterprisev1.S3VolumeSpec) {
+	scopedLog := logC.WithName("ListSmartStoreVolumes")
+	//var temp string
+	for _, volume := range volumeList {
+		scopedLog.Info("Volume: ", "VolumeName: ", volume.RemoteName, "S3EndPoint: ", volume.S3Endpoint)
+	}
+}
+
+// ListSmartStoreIndexes lists smartstore indexes
+func ListSmartStoreIndexes(indexList []enterprisev1.IndexSpec) {
+	scopedLog := logC.WithName("ListSmartStoreIndexes")
+	for _, index := range indexList {
+		scopedLog.Info("Index: ", "IndexName: ", index.Indexname, "RemoteLocation: ", index.RemoteLocation)
+	}
+}
+
+// isSmartstoreEnabled checks and returns true if smartstore is configured
+func isSmartstoreConfigured(smartstore *enterprisev1.SmartStoreSpec) bool {
+	return !(smartstore.IndexList == nil && smartstore.S3VolList == nil)
```

Comment:
what if smartstore == nil?

Reply:
> what if smartstore == nil?

I wanted to verify through debugger once to be sure before I reply. I just checked, and since SmartStore is a struct object, eve if the smartstore spec is not configured, it will not become nil, but the values it holds can be nil.

Reply:
@mikedickey  Verified through the debugger, and smartstore never becomes nil even if we don't specify in the CR. SmartStore is a struct object(not dynamically allocated) in the  StandaloneSpec and IndexerClusterSpec data structures.